snap-core/src/main/java/org/esa/snap/core/util/math/LookupTable.java

```diff
@@ -381,18 +381,30 @@ public final double getValue(final FracIndex[] fracIndexes, final double[] v) {
      *                   value was equal to the partition minimum (maximum).
      * @param fracIndex  the {@link FracIndex}.
      */
-    public final static void computeFracIndex(final IntervalPartition partition, final double coordinate,
-                                 final FracIndex fracIndex) {
+    public static void computeFracIndex(final IntervalPartition partition, final double coordinate,
```

Comment:
How about this? Less code duplication.

```
public static void computeFracIndex(final IntervalPartition partition, final double coordinate, final FracIndex fracIndex) {
        int lo = 0;
        int hi = partition.getCardinal() - 1;

        final boolean isIncreasing = partition.getMonotonicity() > 0;
        while (hi > lo + 1) {
            final int m = (lo + hi) >> 1;

            boolean setHi;
            if (isIncreasing) {
                setHi = coordinate < partition.get(m);
            } else {
                setHi = coordinate > partition.get(m);
            }

            if (setHi) {
                hi = m;
            } else {
                lo = m;
            }
        }

        fracIndex.i = lo;
        fracIndex.f = (coordinate - partition.get(lo)) / (partition.get(hi) - partition.get(lo));
        fracIndex.truncate();
    }
```

Reply:
True. Less code duplication, but the "if" is traversed multiple times, even though it is clear from the very beginning of the loop, which branch is to take, therefore I prefer to leave the code duplication for the sake of logical clearness. One colud implement a comparison operator, but not sure how this would impact performance. Nothing is faster than a primitive comparison, isn't it?

Reply:
I would accept a little performance penalty, if there is one at all, for the clearer code. Indeed, my version is clearer, I think. :-)
The VM would do the performance tweaking.